controllers/validator.go

```diff
@@ -53,8 +52,8 @@ func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error)
 			return false, errors.New("BackupLocation must have cloud storage prefix when backupImages is not set to false")
 		}
 
-		// Check the Velero flags 'no-secret' or 'no-default-backup-location' are not set
-		if !(dpa.Spec.Configuration.Velero.HasFeatureFlag("no-secret") || dpa.Spec.Configuration.Velero.NoDefaultBackupLocation) {
+		// Check if the Velero feature flag 'no-secret' is not set
+		if !dpa.Spec.Configuration.Velero.HasFeatureFlag("no-secret") {
```

Comment:
We do not need to check `NoDefaultBackupLocation` here. If code reaches here, it means `NoDefaultBackupLocation` is false